Khan/react-loadable
# FEI-4763: Throw errors instead of passing them to loading

## Summary:
Errors that aren't caught by vitejs/webpack within a loadable can result in the loadable appearing to be stuck in the loading state.  This is because the API passes the error to the loading component and expects it to check if there's an error or not.  If we actually did this in webapp, it would result in a lot of boilerplate which is annoying.  In order to avoid the boilerplate and bubble up the error to webapp, this PR throws the error instead of passing it to loading.  This is okay b/c webapp has an ErrorBoundary that will catch and log an errors.

Issue: [FEI-4763](https://khanacademy.atlassian.net/browse/FEI-4763)

TODO:
- [x] update the docs in README.md since this changes the API a bit, the section on `Loading error states` will need to be rewritten
- [x] update docs to describe the new `throwOnError` option
- [x] extract jest upgrade and migration to inline snapshots into a separate PR and rebase this one

## Test plan:
### Automated testing:
- yarn test

```
 PASS  __tests__/test.js (6.096s)
  ✓ loading success (417ms)
  ✓ delay and timeout (305ms)
  ✓ loading error (408ms)
  ✓ server side rendering (404ms)
  ✓ server side rendering es6 (407ms)
  ✓ server side rendering waitForLoad/areAllLoaded (402ms)
  ✓ server side rendering es6 waitForLoad/areAllLoaded (403ms)
  ✓ preload (403ms)
  ✓ render (404ms)
  ✓ render component with bad import (409ms)
  ✓ loadable map success (404ms)
  ✓ loadable map error (404ms)
  preloadReady
    ✓ undefined (203ms)
    ✓ one (202ms)
    ✓ many (202ms)
    ✓ missing (204ms)
    ✓ delay with 0 (302ms)

Test Suites: 1 passed, 1 total
Tests:       17 passed, 17 total
Snapshots:   35 passed, 35 total
Time:        6.52s, estimated 7s
Ran all test suites.
-------------------------------|----------|----------|----------|----------|-------------------|
File                           |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
-------------------------------|----------|----------|----------|----------|-------------------|
All files                      |    85.29 |    70.83 |       88 |    85.07 |                   |
 __fixtures__                  |    85.71 |      100 |       75 |    85.71 |                   |
  component-with-bad-import.js |      100 |      100 |      100 |      100 |                   |
  component.es6.js             |      100 |      100 |      100 |      100 |                   |
  component.js                 |      100 |      100 |      100 |      100 |                   |
  foo.js                       |       50 |      100 |        0 |       50 |                 4 |
 src                           |    85.27 |    70.83 |    89.13 |    85.04 |                   |
  index.js                     |    85.27 |    70.83 |    89.13 |    85.04 |... 75,295,303,319 |
-------------------------------|----------|----------|----------|----------|-------------------|
✨  Done in 7.74s.
```

### Manual testing:
- make the same modifications to services/static/node_modules/react-loadable/lib/index.js in webapp
- delete services/static/node_modules/.vite
- restart vitejs in hotel
- modify about.jsx to include the following line: 'import {ViewBad} from "@khanacademy/wonder-blocks-core";'
- load localhost:8224/about
- see the "Oops. Something went wrong." error rendered (w/o this change you'd see the loading spinner forever)
- see following error logged in the console: 'Uncaught SyntaxError: ambiguous indirect export: ViewBad' (w/o this change there'd be no error logged)
